### ISSUE TITLE
Version 6.1.21 of `@storybook/vue` breaks '*.vue' module types

### ISSUE THREAD

**Issue by F0rsaken** [storybookjs/storybook#14220]
**Describe the bug**
Installing newest version (6.1.21) of `@storybook/vue` in Vue 2 + TypeScript project, breaks module augmentation specified in `shims-vue.d.ts` file. All imports of Vue components in TS files are now typed as `{}`, not `VueContructor`.

**To Reproduce**

1. Create Vue 2 project using `vue-cli`
2. Add storybook, using `npx -p @storybook/cli sb init`
3. Create file where you import component like that:
```ts
import { VueConstructor } from 'vue'

import HelloWorld from '@/components/HelloWorld.vue'

const component: VueContructor = HelloWorld
```
4. File should log error when running app with `yarn serve` (or `npm run serve`)

For me the error message was like that:
```
13:22 Conversion of type '{ BasePagination: {}; }' to type 'Components' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.
  Property 'BasePagination' is incompatible with index signature.
    Type '{}' is missing the following properties from type 'VueConstructor<Vue>': extend, nextTick, set, delete, and 9 more.
```

**Expected behavior**
Adding library should not break existing typings

**Screenshots**
Added console error above

**Code snippets**
Added in example

**System**
Please paste the results of `npx sb@next info` here.

```
Environment Info:

  System:
    OS: macOS 10.15.7
    CPU: (4) x64 Intel(R) Core(TM) i5-7360U CPU @ 2.30GHz
  Binaries:
    Node: 14.12.0 - ~/.nvm/versions/node/v14.12.0/bin/node
    Yarn: 1.22.5 - ~/.nvm/versions/node/v14.12.0/bin/yarn
    npm: 6.14.8 - ~/.nvm/versions/node/v14.12.0/bin/npm
  Browsers:
    Chrome: 89.0.4389.82
    Firefox: 79.0
    Safari: 14.0.1
  npmPackages:
    @storybook/addon-actions: ^6.1.21 => 6.1.21 
    @storybook/addon-essentials: ^6.1.21 => 6.1.21 
    @storybook/addon-links: ^6.1.21 => 6.1.21 
    @storybook/vue: ^6.1.21 => 6.1.21 
```

**Comment by shilman** [comment]
@F0rsaken do you know when the bug was introduced? is it still a problem on 6.2-beta?

**Comment by F0rsaken** [comment]
@shilman should the `6.2-beta` work with Vue 2? I could test it later

**Comment by shilman** [comment]
Yes. In the beta, `@storybook/vue` is still vue2 and `@storybook/vue3` is vue3.

**Comment by F0rsaken** [comment]
@shilman yes, unfortunately, the same problem exists in newest `v6.2.0-beta`

**Comment by Cyclodex** [comment]
@shilman FYI With latest `6.2.0-rc.3` i have the same issues
```
Environment Info:

  System:
    OS: Windows 10 10.0.19042
    CPU: (8) x64 Intel(R) Xeon(R) CPU E5-1620 v4 @ 3.50GHz 
  Binaries:
    Node: 14.15.1 - C:\Program Files\nodejs\node.EXE       
    Yarn: 1.3.2 - C:\Program Files (x86)\Yarn\bin\yarn.CMD 
    npm: 6.13.7 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Chrome: 89.0.4389.90
    Edge: Spartan (44.19041.423.0), Chromium (89.0.774.54) 
  npmPackages:
    @storybook/addon-essentials: ^6.2.0-rc.3 => 6.2.0-rc.3 
    @storybook/addon-links: ^6.2.0-rc.3 => 6.2.0-rc.3      
    @storybook/vue: ^6.2.0-rc.3 => 6.2.0-rc.3
```
I thought, if really the @types/react is the problem, then a `npm ls @types/react` might be a good point to see who calls that in the deps.:
```
`-- @storybook/addon-essentials@6.2.0-rc.3
  +-- @storybook/addon-actions@6.2.0-rc.3
  | `-- @storybook/components@6.2.0-rc.3
  |   +-- @types/react-color@3.0.4
  |   | +-- @types/react@17.0.3  deduped
  |   | `-- @types/reactcss@1.2.3
  |   |   `-- @types/react@17.0.3  deduped
  |   `-- @types/react-syntax-highlighter@11.0.5
  |     `-- @types/react@17.0.3  deduped
  +-- @storybook/addon-docs@6.2.0-rc.3
  | `-- @storybook/builder-webpack4@6.2.0-rc.3
  |   `-- @storybook/ui@6.2.0-rc.3
  |     `-- @types/markdown-to-jsx@6.11.3
  |       `-- @types/react@17.0.3  deduped
  `-- @storybook/api@6.2.0-rc.3
    `-- @types/reach__router@1.3.7
      `-- @types/react@17.0.3
```

**Comment by Cyclodex** [comment]
@shilman forget about my comment above - I figured out why my vue build and storybook was throwing errors 😎 

As one can see in the output above, I had an `shims-tsx.d.ts` file in my `src/` folder, which we don't actually need.
(Where it has a conflict with the `interface Element extends VNode {}` with something coming from the storybook dependencies)

So I deleted this file, and now all is fine! Sorry for the confusion :) Cheers

**Comment by vanessayuenn** [comment]
Hi! I am going through and cleaning up old issues in the repo. We just shipped a ton of improvements in Storybook 7.0 (especially with vue!), and I suspect this bug might be fixed already. Can you please try it out? If this issue persists with the latest version, please open a new bug report with a 7.x reproduction. Thank you so much! 🙏🏼
